web_widget_auto_color/static/src/js/view_list.js

```diff
@@ -0,0 +1,61 @@
+        },
+
+        getIntValue: function(str){
+            var sum = 0
+            for (i=0; i<str.length;i++){
+                sum += str.charCodeAt(i);
+            }
+            return sum
+        },
+
+        auto_color_cell_style: function(record, column){
+            style = ''
+            value = record.get(column.name)
+            if (value != false && value != undefined) {
+                var intValue = this.getIntValue(value)
+                colors = this.get_colors()
+                bgcolor = colors[intValue % colors.length]
```

Comment:
It can be improved by generating the color directly with something like this: http://stackoverflow.com/questions/8132081/generate-a-random-seeded-hex-color
And use intValue as a seed.